components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -604,6 +616,64 @@ protected String getSubjectFromUserIDClaimURI(AuthenticationContext context) {
         return subject;
     }
 
+    protected String getSubjectFromUserIDClaimURI(AuthenticationContext context, Map<String, Object> idTokenClaims)
+            throws AuthenticationFailedException {
+
+        boolean useLocalClaimDialect = context.getExternalIdP().useDefaultLocalIdpDialect();
+        String userIdClaimUri = context.getExternalIdP().getUserIdClaimUri();
+        String spTenantDomain = context.getTenantDomain();
+
+        try {
+            if (useLocalClaimDialect) {
+                // User ID is defined in local claim dialect at the IDP. Find the corresponding OIDC claim and retrieve
+                // from idTokenClaims.
+                String userIdClaimUriInOIDCDialect = getUserIdClaimUriInOIDCDialect(userIdClaimUri, spTenantDomain);
+                return (String) idTokenClaims.get(userIdClaimUriInOIDCDialect);
+
+            } else {
+                ClaimMapping[] claimMappings = context.getExternalIdP().getClaimMappings();
+                // Try to find the userIdClaimUri within the claimMappings.
+                if (!ArrayUtils.isEmpty(claimMappings)) {
+                    for (ClaimMapping claimMapping : claimMappings) {
+                        if (StringUtils.equals(claimMapping.getRemoteClaim().getClaimUri(), userIdClaimUri)) {
+                            // Get the subject claim in OIDC dialect.
+                            String userIdClaimUriInLocalDialect = claimMapping.getLocalClaim().getClaimUri();
+                            return (String) idTokenClaims
+                                    .get(getUserIdClaimUriInOIDCDialect(userIdClaimUriInLocalDialect, spTenantDomain));
+                        }
+                    }
+                }
+            }
+        } catch (ClaimMetadataException ex) {
+            throw new AuthenticationFailedException(
+                    "Error while executing claim transformation for IDP: " + context.getExternalIdP().getIdPName(), ex);
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("Couldn't find the subject claim from the idToken claims.");
```

Comment:
Couldn't find the subject claim from the idToken claims. --> Couldn't find the subject claim among id_token claims.

Also can we add some context info like idpName?